tsdb/batcher.go

```diff
@@ -117,8 +117,12 @@ func (b *PointBatcher) Stop() {
 		return
 	}
 
-	close(b.stop)
-	b.wg.Wait()
+	// If not stopped, stop the batching go routine gracefully
+	if b.stop != nil {
+		b.stop <- struct{}{}
```

Comment:
This should be the equivalent of closing the channel to my knowledge. A closed channel always returns the default value and will always be chosen in a `select`. Does that not work correctly?